documentation/dataplane/dash-flow/DASH_FLOW_API_HLD.md

```diff
@@ -0,0 +1,982 @@
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_FLOW_TABLE_ATTR_TABLE_EXPIRE_TIME,
+
+    /**
+     * @brief Action flow_table_action parameter TABLE_VERSION
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_FLOW_TABLE_ATTR_TABLE_VERSION,
+
+    /**
+     * @brief Action flow_table_action parameter TABLE_KEY_FLAG
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_FLOW_TABLE_ATTR_TABLE_KEY_FLAG,
+
+    /**
+     * @brief Action flow_table_action parameter TABLE_TCP_TRACK_STATE
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     */
+    SAI_FLOW_TABLE_ATTR_TABLE_TCP_TRACK_STATE,
+
+    /**
+     * @brief Action flow_table_action parameter TABLE_TCP_RESET_ILLEGAL
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     */
+    SAI_FLOW_TABLE_ATTR_TABLE_TCP_RESET_ILLEGAL,
+
+    /**
+     * @brief End of attributes
+     */
+    SAI_FLOW_TABLE_ATTR_END,
+
+    /** Custom range base value */
+    SAI_FLOW_TABLE_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_FLOW_TABLE_ATTR_CUSTOM_RANGE_END,
+
+} sai_flow_table_attr_t;
+
+/**
+ * @brief Create dash_flow_flow_table
+ *
+ * @param[out] flow_table_id Entry id
+ * @param[in] switch_id Switch id
+ * @param[in] attr_count Number of attributes
+ * @param[in] attr_list Array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t (*sai_create_flow_table_fn)(
+        _Out_ sai_object_id_t *flow_table_id,
+        _In_ sai_object_id_t switch_id,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list);
+
+/**
+ * @brief Remove dash_flow_flow_table
+ *
+ * @param[in] flow_table_id Entry id
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t (*sai_remove_flow_table_fn)(
+        _In_ sai_object_id_t flow_table_id);
+
+/**
+ * @brief Set attribute for dash_flow_flow_table
+ *
+ * @param[in] flow_table_id Entry id
+ * @param[in] attr Attribute
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t (*sai_set_flow_table_attribute_fn)(
+        _In_ sai_object_id_t flow_table_id,
+        _In_ const sai_attribute_t *attr);
+
+/**
+ * @brief Get attribute for dash_flow_flow_table
+ *
+ * @param[in] flow_table_id Entry id
+ * @param[in] attr_count Number of attributes
+ * @param[inout] attr_list Array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t (*sai_get_flow_table_attribute_fn)(
+        _In_ sai_object_id_t flow_table_id,
+        _In_ uint32_t attr_count,
+        _Inout_ sai_attribute_t *attr_list);
+
+```
+
+## Flow APIs
+
+### Basic flow APIs
+
```

Comment:
same as above, for each attribute, it will be better to have them explained using table.